pkg/cloudproviders/gcp/storage/handler.go

```diff
@@ -48,7 +48,18 @@ func NewClientHandler(ctx context.Context, creds *google.Credentials) (ClientHan
 func (s *clientHandlerImpl) UpdateClient(ctx context.Context, creds *google.Credentials) error {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
-	s.wg.Wait()
+	waitCh := make(chan struct{})
+	go func() {
+		s.wg.Wait()
+		close(waitCh)
+	}()
+
+	select {
+	case <-ctx.Done():
+		close(waitCh)
+		return ctx.Err()
+	case <-waitCh:
+	}
```

Comment:
MIght be better suited to use:
- a `concurrency.WaitGroup` instead of `sync.WaitGroup`
- then use a `concurrency.WaitInContext` using the `concurrency.WaitGroup` and the `context` here

This way we streamline our concurrency usage a bit more, and it reads much cleaner than the current code IMO.

Then this becomes:
```go
if !concurrency.WaitInContext(s.wg, ctx) {
    return ctx.Err()
}
...
```